Doesn't build with gcc 5

Hi,
The Isengard branch fail to build with gcc 5.2.1 and arches different than i386 or amd64 (armhf, powerpc, armel)



src/uri.cpp:44:3: error: narrowing conversion of ‘(uri::char_class_e)-2’ from ‘int’ to ‘char’ inside { } [-Wnarrowing]
src/uri.cpp:65:3: error: narrowing conversion of ‘(uri::char_class_e)-2’ from ‘int’ to ‘char’ inside { } [-Wnarrowing]
src/uri.cpp:86:3: error: narrowing conversion of ‘(uri::char_class_e)-2’ from ‘int’ to ‘char’ inside { } [-Wnarrowing]
src/uri.cpp:107:3: error: narrowing conversion of ‘(uri::char_class_e)-2’ from ‘int’ to ‘char’ inside { } [-Wnarrowing]
src/uri.cpp:128:3: error: narrowing conversion of ‘(uri::char_class_e)-2’ from ‘int’ to ‘char’ inside { } [-Wnarrowing]

--- a/src/uri.h
+++ b/src/uri.h
@@ -21,13 +21,13 @@
 namespace uri
 {
   /// Char class.
-  enum char_class_e
+  typedef enum char_class_e : signed char
   {
     CINV = -2, ///< invalid
     CEND = -1, ///< end delimitor
     CVAL = 0, ///< valid any position
     CVA2 = 1, ///< valid anywhere but 1st position
-  };
+  } char_class_e_type;
 
   /// Traits used for parsing and encoding components.
   struct traits
@@ -35,7 +35,7 @@ namespace uri
     char* begin_cstring; ///< begin cstring (or 0 if none)
     char begin_char; ///< begin char (or 0 if none)
     char end_char; ///< end char (or 0 if none)
-    char char_class[256]; ///< map of char to class
+    char_class_e_type char_class[256]; ///< map of char to class
   };
 
   /**

